### DISCUSSION TITLE
Using Jamulus for hybrid live and online rehearsals/concerts?

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1800] [General]
About a year ago, after days or weeks of internet research aka ("HOW DO I GET RID OF THIS STUPID LATENCY ON ZOOM??"), I finally found out about Jamulus.

I'm probably not the only one who says this, but this app has been a lifesaver for my choir until now. For sure, not everything went well right away, but eventually we started being able to do real rehearsals over the net. Since February 2021 we're using Jamulus for our weekly rehearsals, did 2 streaming concerts and recorded hours of songs. Our choir organization now runs a semi-public directory server to enable this experience for multiple other choirs in our region. 

Luckily, the numbers of COVID infections are decreasing and multiple choirs are now thinking of returning back to in-person rehearsals. Therefore - at least to me who still wants to use Jamulus - another big topic will come up: 

_How can we combine in-person rehearsals with online Jamulus rehearsals?_
I'm sure a lot of people are interested in this topic - so am I ;-). I'd be happy to have this discussion focus on experiences and knowledge exchange.

Therefore I'll first share my experience here:

A few days ago we connected our church-PA system to a PC which was connected to the internet by 4G/5G. All musicians except one were at home. Surprisingly, the person in church could play almost in time with the online group (overall delay was < 80ms). Now the question is: what's an easy and inexpensive way to capture sound from ~ 10 singers in church (who are at a large distance) without provoking feedback? I'm quite sure that we could provide everyone with a microphone and headphones; however that's not really what we want. In a perfect setting, having loudspeakers and one/two microphones placed in such a way that the live people hear the online people while also the other way round works would be great. I'm sure that this is possible since on stage this must also work somehow. Does anybody have experiences in this field?

**Comment by passing** [comment]
> having loudspeakers and one/two microphones placed in such a way that the live people hear the online people while also the other way round works would be great. I'm sure that this is possible since on stage this must also somehow work. Does anybody have experiences in this field?

The most basic setup (independent of Jamulus) would be one singer, one microphone, one monitor (speaker). The speaker is directed towards the singer to hear himself, but no feedback occurs as the microphone is 'directed' towards the singer (see "Directivity" = "Richtcharakteristik") and so picks up just little sound from the speaker. Feeding an additional signal (coming from Jamulus) into the Speaker should not make a difference.
The situation changes if you want to have just 2 microphones for ~10 Singers (that keep quite some distance), as these Microphones need to be more sensitive to sound from several directions. So they will definitely pick up more sound from the monitor speakers, which will probably make feeback way more likely. I just cannot tell if that's more like a "may work" or a "will fail for sure" or a "it depends", as I have just little experience in this area. But maybe some real audio engineer can jump in here..

**Comment by gene96817** [comment]
I'll want to follow your use case closely. I am working on a similar use case.

**Comment by Bruno79260** [comment]
You could setup 
* a unique central sound capture using either a stereo mike pair or a stereo mike, or even a recorder like the Zoom H5 or H6 like used frequently for recording music concerts, the output of which being then feeded into Jamulus
*  an individual return channel using one headset per singer feeded by the Jamulus out signal. It would mean using an multichannel headset amplifier i.e. Behringer HA8000 V2 or Swissonic Headamp 8.

You could also try having return loudspeakers at your feets, but as said above, the probability of a sound loop would then be very high and very difficult to manage/avoid. 

I think option one (one mike, several headsets) would be more cost effective and much easier to manage.

**Reply by ann0see** [reply]
Thank you very much for your comment. I think we own a Tascam recorder and headsets _might_ be available too. (But hygiene questions might arise since they might need to be shared)

The headset amp doesn’t seem to be too expensive. I haven’t looked at the PA system yet, but maybe we already have one.

**Comment by dakhubgit** [comment]
A bunch of wireless RF headphones using FM?  They don't cause relevant additional delays (must _not_ be digital), can all work off the same sender (assuming they are of the same type or at least use the same frequency/modulation) and don't create extensive cabling traps.  And people can take them home and use them there as well.  The ones intended for TV use may employ excessive AGC.  They tend to be workable, but if there is some transient from somewhere (someone stumbling over the microphone stand), it takes some time before volume picks up again.

**Reply by gene96817** [reply]
It seems that it would be more practical to use a microphone and monitor (speaker) configuration common for performances. In this case, the key is for the microphone not to pick up the monitor sound (which otherwise would result in feedback). 

I won't be able to test this for a couple more months when my chorus begins some in-person rehearsals.

**Reply by Bruno79260** [reply]
It would yes be more comfortable, but while it’s feasible for an individual singing over a dynamic proximity microphone, it’s quite impossible in a configuration where the microphone is of condenser type I.e large membrane that’s necessary for recording a group.
In that situation, the microphone is distant from the choir which hit self produces a lot of harmonics and the room generates acoustical reverberation.

In other word, a cloud of sounds fill’s in the whole space, which is captured by the central microphone and send toward the loudspeakers, making it impossible to avoid a loop.

The option then, if the loudspeaker option is preferred, is to replace the central microphone by as much individual dynamic microphones as singers in the choir. It would require more costly equipments (mixer, XLR câbles, many microphones and feet’s, probably several return loudspeakers) than the solution with one central sound capture and individual headphones.

**Reply by ann0see** [reply]
>  by as much individual dynamic microphones as singers in the choir

Yes, that was a solution I thought of too. I think using wireless headphones or even better wireless receivers in which you can plug in standard consumer headphones are a much easier solution.

**Reply by ann0see** [reply]
Actually I own a wireless headphone made by Sony. The quality is not that great and has too much background noise. I‘m sure new ones are much better. Do you have any recommendation? Would you use one sender and then use multiple headphones on the same channel?

**Reply by dakhubgit** [reply]
Well, I am using Philips wireless headphones mostly advertised for TV use.  I have an RS130 and an RS119.  I usually use the RS130 headphone (it fits more reliably and finds the sender channel automatically) with the RS119 sender (which is called TR119 on its own).  The TR130 sender has a "surround" sound mode that actually works pretty well in connection with the volume pan typical for Mono-in/Stereo-out in Jamulus but its AGC action (which happens only in the senders) seems to be more aggressive, so I tend to use the simpler sender.  I've not done extensive side-by-side comparisons: that was more of a gut impression.

One trick with the AGC is to turn up volume on the headphone amp high enough that the AGC does not have a lot of breathing room.

I am using this with an accordion orchestra which has large dynamics so it's sort of convenient that I don't get the full volume range right next to my ear.

The RF noise level (admittedly with the sender rather close) is low enough that I can pinpoint less and more noisy external microphones.

It's not a HiFi listening experience.  It works for me for rehearsal purposes.  Not as much a recommendation as something that may be worth trying because it's cheap enough (those headphones use exchangeable AAA NiMH batteries so they don't become useless when their batteries age).  Some day I'll need to compare to a Sony or whatever else.

**Reply by ann0see** [reply]
Hmm. These headphones seem to be available on eBay at a reasonably low price. Thanks for your recommendation! 

If we’re lucky, there are some headsets but I‘m still a bit worried about sharing them.

**Reply by dakhubgit** [reply]
Well, if you get some of those, be sure to check the instructions (on Philips' web page) for changing the batteries (one has to twist the pads off in a certain manner rather than just pulling) rather than breaking off notches while thinking that it can't be that hard.  Replacement pads, while available from Philips itself for cheaper than what Ebay offers, tend to ruin the value/price ratio rather thoroughly.  Fortunately breaking off just one notch retains enough stability to have the pads fall off only occasionally.

Just saying.  As I said: the "recommendation" is more in the "works for me" rather than "made for that purpose" category, enough so that I forego the good wired headphones most of the time.  It's also convenient that I can set up the server and then go elsewhere until someone actually appears (in which case I have to get back to the vicinity of the microphone).  Or leave for a moment without missing the action.

**Reply by dakhubgit** [reply]
The main problem with wireless these days is that you _have_ to get the old ones with analog technology and FM (and consequently a limited SNR ratio).  Go digital (like Bluetooth), and your system delay goes up.  If you want to support a large number of people, sourcing a large number of them may become tricky.

**Reply by ann0see** [reply]
Just talked to one of our responsible persons and they‘d rather just use Zoom to stream one way audio & video for now 😒 (while letting the Jamulus people rehearse online at another time). 

I‘m still in favour of adding some part of Jamulus in these rehearsals.

**Reply by dakhubgit** [reply]
I'm with you.  We're going back to live rehearsals next week, and, well, I am not sure how to sell a few things that have proven useful.  Like being able to put in a drum track (either default arranger stuff or stuff I typed into a notation program) at arbitrary speed and entirely unimpressed by other players.  Or the ability to turn up the volume of the bass accordion for purposes of keeping tight timing.

Either of which can be amended by equipment I could store next to our rehearsal room (I think the bass accordion has a built-in microphone that would be comparatively proof against feedback).  I have a nice 100W keyboard amp from the seventies in washing machine size and it doesn't use bass reflex tubes or similar nonsense to move air.  It does a pretty solid bass drum impression and can still put up an undistorted drum kit (and likely a bass accordion) on top without sounding anything like a "car HiFi".

Still likely a harder sell than if I can just inject the arranger into my mixer strip without needing an amp.

Geek time may be setting.

**Reply by ann0see** [reply]
Yeah, there’s a lot of stuff possible with tech nowadays ;-). Although we‘ll probably not get rid of Jamulus (we still have huge plans with the software) it will become less important. 

 I was actually thinking of getting a bunch of choirs/musicians together in some kind of summer concert; but I didn’t have time to do planning or anything else.

**Reply by gene96817** [reply]
> The main problem with wireless these days is that you have to get the old ones with analog technology and FM (and consequently a limited SNR ratio). Go digital (like Bluetooth), and your system delay goes up. If you want to support a large number of people, sourcing a large number of them may become tricky.

The rules might vary in different (wireless world) regions. In North America, the spectrum for the old analog technology was being "retired" for new (digital) applications. This means that if you are using the old analog technology, you should expect the channel to become noisier as the new generation equipment comes to market. (This is part of the reassignment of spectrum white space.)

**Comment by Andrew999999** [comment]
Slightly late to this discussion - ann0see how did you get on in the end?  One thing that occurred to me was to use smartphones with Jamulus android client as your "wireless headphone" device.  With a wired headest/earpiece this gives reasonably low latency in my experience.  Using a smartphone microphone may also be feasible - my testing was unsatisfactory as they appear to be tuned for voice - but for singing this may be fine.  So in effect every singer could have their own personal mic/headset.

**Reply by ann0see** [reply]
Unfortunately we used Zoom ;-(.
Maybe in future something like the headphone approach might work.

How did you connect the smartphones WiFi doesn’t work well on them.

**Reply by Andrew999999** [reply]
In my testing I used an ethernet USB-C on-the-go (OTG) adapter.  These are very cheap.  This allowed me to connect the smartphone to an ethernet switch - although this could have been the Internet router.  I got performance comparable to running jamulus on a wired PC on the same network.  You're right - wi-fi not only introduced considerable lag, but it was also very variable and unpredictable (from bad to awful).  I then plugged the over-ear headphones into the smartphone.

![31rP3YZFGvL _AC_](https://user-images.githubusercontent.com/66430820/130368513-0216eab7-eec0-47fd-8d9f-16394e3b16a2.jpg)
